### PR TITLE
dosemu2: init at git 2020-11-04

### PR DESCRIPTION
**Pull request by cosarara**
###### Motivation for this change

I wanted to run dosemu2 on nixos so I made a package.

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/misc/emulators/dosemu2/default.nix [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+    runHook postUnpack
+  '';
+
+  postUnpack = ''
+        sed -i -e "s|DATE=.*|DATE='Mon, 2 Nov 2020 15:49:15 +0300'|" "$sourceRoot/git-rev.sh"
+        sed -i -e "s:^}$:  prefix $prefix\n&:g" "$sourceRoot/compiletime-settings"
+  '';
+
+  preConfigure = ''
+        ./autogen.sh
```
Comment (SuperSandro2000) [inline pkgs/misc/emulators/dosemu2/default.nix:56]:
```suggestion
    ./autogen.sh
```

Nix uses 2 tabs to intend things.

File: pkgs/misc/emulators/dosemu2/default.nix [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+  '';
+
+  installPhase = ''
+        if [ -n "$prefix" ]; then
+          mkdir -p "$prefix";
+        fi;
+        make SHELL="$(which bash)" PREFIX="$prefix" install
+
```
Comment (SuperSandro2000) [inline pkgs/misc/emulators/dosemu2/default.nix:64]:
```suggestion
```

File: pkgs/misc/emulators/dosemu2/default.nix [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+  '';
+
+  meta = with stdenv.lib; {
+    description = "DOS emulator for linux";
+    homepage = "http://dosemu2.github.io/dosemu2/";
+    license = licenses.gpl2Only;
```
Comment (SuperSandro2000) [inline pkgs/misc/emulators/dosemu2/default.nix:70]: On which platforms does that work? I guess linux only.

File: pkgs/misc/emulators/dosemu2/default.nix [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+    maintainers = [ "Jaume Delclòs Coll <jaume@delclos.com>" ];
```
Comment (SuperSandro2000) [inline pkgs/misc/emulators/dosemu2/default.nix:71]: Please add yourself to maintainers.nix and reference your name here.

**Comment by SuperSandro2000** [comment]
Result of `nixpkgs-review pr 102819` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>dosemu2</li>
  </ul>
</details>

File: pkgs/misc/emulators/dosemu2/default.nix (hunk 1 above)
Comment (cosarara) [reply pkgs/misc/emulators/dosemu2/default.nix]: Fixed!

File: pkgs/misc/emulators/dosemu2/default.nix (hunk 2 above)
Comment (cosarara) [reply pkgs/misc/emulators/dosemu2/default.nix]: Fixed!

File: pkgs/misc/emulators/dosemu2/default.nix (hunk 3 above)
Comment (cosarara) [reply pkgs/misc/emulators/dosemu2/default.nix:69]: As far as I know, yes.

File: pkgs/misc/emulators/dosemu2/default.nix (hunk 4 above)
Comment (cosarara) [reply pkgs/misc/emulators/dosemu2/default.nix]: Done.

File: pkgs/misc/emulators/dosemu2/default.nix (hunk 3 above)
Comment (SuperSandro2000) [reply pkgs/misc/emulators/dosemu2/default.nix:69]: Can you add ``platforms = platforms.linux`` to meta?

File: pkgs/misc/emulators/dosemu2/default.nix [hunk 5]
```diff
@@ -0,0 +1,73 @@
+{ stdenv, fetchFromGitHub, fetchurl,
+bison, flex, SDL2, udev, libslirp, json_c, autoconf, automake,
+pkg-config, libbsd, hexdump, slang, xorg, readline, libao,
+gpm, munt, libieee1284 }:
+stdenv.mkDerivation {
```
Comment (SuperSandro2000) [inline pkgs/misc/emulators/dosemu2/default.nix:5]:
```suggestion
{ stdenv, fetchFromGitHub, fetchurl
, bison, flex, SDL2, udev, libslirp, json_c, autoconf, automake
, pkg-config, libbsd, hexdump, slang, xorg, readline, libao
, gpm, munt, libieee1284 }:

stdenv.mkDerivation {
```

File: pkgs/misc/emulators/dosemu2/default.nix [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+  name = "dosemu2";
```
Comment (SuperSandro2000) [inline pkgs/misc/emulators/dosemu2/default.nix:6]:
```suggestion
  pname = "dosemu2";
  version = "X.X.X"
```

Please specify a version.

File: pkgs/misc/emulators/dosemu2/default.nix [hunk 7]
```diff
@@ -0,0 +1,73 @@
...
+    slang xorg.mkfontdir readline libao gpm munt
+    libieee1284
+  ];
+
+  configureFlags = [
+    "--disable-fdpp"
+    "--with-fdtarball=dosemu-freedos-1.0-bin.tgz"
+  ];
+
+  unpackPhase = ''
```
Comment (SuperSandro2000) [inline pkgs/misc/emulators/dosemu2/default.nix:32]: Could you do this in a pre or postPhase?

File: pkgs/misc/emulators/dosemu2/default.nix [hunk 8]
```diff
@@ -0,0 +1,73 @@
...
+  '';
+
+  preConfigure = ''
+    ./autogen.sh
+  '';
+
+  installPhase = ''
+    if [ -n "$prefix" ]; then
+      mkdir -p "$prefix";
+    fi;
```
Comment (SuperSandro2000) [inline pkgs/misc/emulators/dosemu2/default.nix:62]:
```suggestion
    mkdir -p "$prefix";
```

File: pkgs/misc/emulators/dosemu2/default.nix [hunk 9]
```diff
@@ -0,0 +1,73 @@
...
+    make SHELL="$(which bash)" PREFIX="$prefix" install
```
Comment (SuperSandro2000) [inline pkgs/misc/emulators/dosemu2/default.nix:63]:
```suggestion
    make PREFIX="$prefix" install
```
This should not be required.

File: pkgs/misc/emulators/dosemu2/default.nix [hunk 10]
```diff
@@ -0,0 +1,73 @@
...
+  '';
+
+  meta = with stdenv.lib; {
+    description = "DOS emulator for linux";
+    homepage = "http://dosemu2.github.io/dosemu2/";
+    license = licenses.gpl2Only;
+    maintainers = [ maintainers.cosarara ];
```
Comment (SuperSandro2000) [inline pkgs/misc/emulators/dosemu2/default.nix:70]:
```suggestion
    maintainers = with maintainers; [ cosarara ];
```

**Comment by SuperSandro2000** [comment]
Result of `nixpkgs-review pr 102819` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>dosemu2</li>
  </ul>
</details>

File: pkgs/misc/emulators/dosemu2/default.nix (hunk 9 above)
Comment (cosarara) [reply pkgs/misc/emulators/dosemu2/default.nix:63]: IIRC it does not build without it. Some makefile down the line has bashisms in it.

File: pkgs/misc/emulators/dosemu2/default.nix (hunk 6 above)
Comment (cosarara) [reply pkgs/misc/emulators/dosemu2/default.nix:6]: The source is a recent git commit, not a released version.

Comment (SuperSandro2000) [reply pkgs/misc/emulators/dosemu2/default.nix:6]:
```suggestion
  pname = "dosemu2";
  version = "unstable-XXXX-XX-XX";
```
Then please set version to the date of the commit.

**Comment by cosarara** [comment]
I am not using nixos anymore, therefore I will not be maintaining this. Thanks for taking the time to review it so far.
If anybody wants to adopt it, of course feel free.

**Pull request closed without merging**